tests/tls_e2e/host/host.cpp

```diff
@@ -216,52 +220,32 @@ void* client_thread(void* arg)
         // g_client_thread_exit_code could be any values in negative test cases
         g_client_thread_exit_code = 0;
     }
-    pthread_exit((void*)&g_client_thread_exit_code);
+
     fflush(stdout);
 done:
     return NULL;
 }
 
 int run_test_with_config(tls_test_configs_t* test_configs)
 {
-    pthread_attr_t server_tattr;
-    pthread_attr_t client_tattr;
-    pthread_t client_thread_id;
-    int ret = 0;
-    void* retval = NULL;
-
     // create server thread
-    ret = pthread_attr_init(&server_tattr);
-    if (ret)
-        oe_put_err("pthread_attr_init(server): ret=%u", ret);
-
-    ret = pthread_create(
-        &server_thread_id, NULL, server_thread, (void*)test_configs);
-    if (ret)
-        oe_put_err("pthread_create(server): ret=%u", ret);
+    server_thread = std::thread(run_server, (void*)test_configs);
 
     OE_TRACE_INFO("wait until TLS server is ready to accept client request\n");
-    pthread_mutex_lock(&server_mutex);
+    std::unique_lock<std::mutex> l(server_mutex);
     while (!g_server_condition)
-        pthread_cond_wait(&server_cond, &server_mutex);
-    pthread_mutex_unlock(&server_mutex);
+        server_cond.wait(l);
```

Comment:
Can you double-check the logic here? Previously, upon return from pthread_cond_wait, the lock was immediately unlocked via a pthread_mutex_unlock.
In this new version, we don't seem to unlock the mutex immediately upon return from wait.

Reply:
Oh, you're right, I misread the `condition_variable::wait` and thought the lock was released when the condition was met, but thats not the case. I'll fix